tests/phpUnit/Translation/TranslationDelegateTest.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Translation;
+
+use App\Translation\TranslationApiInterface;
+use App\Translation\TranslationDelegate;
+use App\Translation\TranslationResult;
+use InvalidArgumentException;
+use PHPUnit\Framework\TestCase;
+use Throwable;
+
+/**
+ * @internal
+ * @covers \App\Translation\TranslationDelegate
+ */
+class TranslationDelegateTest extends TestCase
+{
+  public function testSingleApi(): void
+  {
+    $api = $this->createMock(TranslationApiInterface::class);
+    $expected_result = new TranslationResult();
+    $api->expects($this->once())
+      ->method('translate')
+      ->willReturn($expected_result)
+    ;
+
+    $translation_delegate = new TranslationDelegate($api);
+
+    $actual_result = $translation_delegate->translate('test', 'en', 'fr');
+
+    $this->assertEquals($expected_result, $actual_result);
+  }
+
+  public function testTryNextApiIfFirstFails(): void
+  {
+    $api1 = $this->createMock(TranslationApiInterface::class);
+    $api2 = $this->createMock(TranslationApiInterface::class);
+
+    $api1->expects($this->once())
+      ->method('translate')
+      ->willReturn(null)
+    ;
+
+    $expected_result = new TranslationResult();
+    $api2->expects($this->once())
+      ->method('translate')
+      ->willReturn($expected_result)
+    ;
+
+    $translation_delegate = new TranslationDelegate($api1, $api2);
+
+    $actual_result = $translation_delegate->translate('test', 'en', 'fr');
+
+    $this->assertEquals($expected_result, $actual_result);
+  }
+
+  public function testAllApiFails(): void
+  {
+    $api1 = $this->createMock(TranslationApiInterface::class);
+    $api2 = $this->createMock(TranslationApiInterface::class);
+
+    $api1->expects($this->once())
+      ->method('translate')
+      ->willReturn(null)
+    ;
+
+    $api2->expects($this->once())
+      ->method('translate')
+      ->willReturn(null)
+    ;
+
+    $translation_delegate = new TranslationDelegate($api1, $api2);
+
+    $result = $translation_delegate->translate('test', 'en', 'fr');
+
+    $this->assertNull($result);
+  }
+
+  public function testInvalidLanguageCode(): void
```

Comment:
Maybe add a test for a few valid language codes as well

Reply:
testSingleApi and others all test with valid language code. Is that what you meant?

Reply:
I didn't see that the other tests used valid language codes so an extra test is not needed. I'll resolve the comment